### PR TITLE
Scales Department Access to Pop and Fixes Access Rights

### PR DESCRIPTION
**Pull request by ZomgPonies**
## What Does This PR Do
Adds another config to define at how many players to switch from department access to regular access rights.

Tuned the following access rights during department access:

Engie - Removed Head, Keycard and Sec Door access
Science - Removed AI Upload, Head, Keycard and Sec Door access
Medical - Removed Head, Keycard and Sec Door access
Security - Everyone but Warden loses Head and Keycard access.
Support - Everyone but Bartender, QM and Bouncer loses Weapon Access

## Why It's Good For The Game
Automating switching from LowPop to MediumPop and vice-versa removes risks of admin mistakes. 

## Changelog
:cl:
add: department access population scaling
tweak: tweaked access permissions during department access
/:cl:

### CODE REVIEW

File: code/game/jobs/job/engineering.dm [hunk 1]
```diff
@@ -59,7 +59,7 @@
 	supervisors = "the chief engineer"
 	department_head = list("Chief Engineer")
 	selection_color = "#fff5cc"
-	department_access = list(ACCESS_ATMOSPHERICS, ACCESS_CE, ACCESS_CONSTRUCTION, ACCESS_EMERGENCY_STORAGE, ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_EVA, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_HEADS, ACCESS_KEYCARD_AUTH, ACCESS_MAINT_TUNNELS, ACCESS_MECHANIC, ACCESS_MINERAL_STOREROOM, ACCESS_MINISAT, ACCESS_RC_ANNOUNCE, ACCESS_SEC_DOORS, ACCESS_TCOMSAT, ACCESS_TECH_STORAGE, ACCESS_TELEPORTER)
+	department_access = list(ACCESS_ATMOSPHERICS, ACCESS_CE, ACCESS_CONSTRUCTION, ACCESS_EMERGENCY_STORAGE, ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_EVA, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_MAINT_TUNNELS, ACCESS_MECHANIC, ACCESS_MINERAL_STOREROOM, ACCESS_MINISAT, ACCESS_RC_ANNOUNCE, ACCESS_TCOMSAT, ACCESS_TECH_STORAGE, ACCESS_TELEPORTER)
```
Comment (trololiver112) [inline code/game/jobs/job/engineering.dm:62]: I do no think any non command should have access to the office of a head, even on lowpop. so this comment goes for most of the departments like for example a doctor with CMO office access.(Which is what i am guessing ACCESSS_CMO is. In my opinion that is.

**Comment by trololiver112** [comment]
another thing i would add to that is i don't think everyone should have access to the mimes, clowns, QMs chaplains offices. Exception being maybe the janitor?

**Comment by blinkdog** [comment]
I apologize for the ham-fisted nature of #123 

I saw a lot of comments on Discord about limited access being a real pain point for play.

I disabled `JOBS_HAVE_MINIMAL_ACCESS` in `config.txt`. While this helped (a little), it switched most jobs to an access set that did NOT include `ACCESS_MAINT_TUNNELS`. The complaints on Discord started to be that limited access and being cut off from maintenance were still pain points for play.

So enter #123, which (more or less) took department head access and gave it to every job in the department.
This was less painful; people could access most areas they needed to get to plus maintenance.
However, I think consensus is that it is too much access; it needs to be tweaked back in the other direction.

I agree with the changes in the PR and @trololiver112 's suggestions about service and command.

**Comment by ZomgPonies** [comment]
Adjusted permissions as per comments, should be good now.

**Comment by ClockworkGolem** [comment]
Is it intended that the chaplain (and presumably other non-food-related service jobs) should have access to Botany on lowpop? Especially considering that most botanists (myself included) will either borrow or build a chem dispenser for that room if possible. Greytiders might take the opportunity to steal hatchets or death nettles, or to intentionally let out bees. Noticed this issue while getting griefed last night. I had expected that only cooks and maybe bartenders would have extra skeleton-crew access to the room.

**Comment by blinkdog** [comment]
> Is it intended that the chaplain (and presumably other non-food-related service jobs) should have access to Botany on lowpop? Especially considering that most botanists (myself included) will either borrow or build a chem dispenser for that room if possible. Greytiders might take the opportunity to steal hatchets or death nettles, or to intentionally let out bees. Noticed this issue while getting griefed last night. I had expected that only cooks and maybe bartenders would have extra skeleton-crew access to the room.

I see what happened here. The original department access gave service jobs cross-job access. Most of the service jobs lived in the code file `code/game/jobs/job/support.dm` that was edited to remove that cross-job access. The one exception? The Chaplain. Their access lives in the file `code/game/jobs/job/support_chaplain.dm` and they only lost `ACCESS_WEAPONS` but retained all the cross-job access.

I have opened #160 so we can fix this up. Thanks for the heads-up, @ClockworkGolem

**Pull request merged**